src/main/java/com/gojek/beast/sink/bq/BqSink.java

```diff
@@ -21,11 +27,33 @@
 
     private final BigQuery bigquery;
     private final TableId tableId;
+    private final BQResponseParser responseParser;
+    private final BQErrorHandler errorHandler; // handler instance for BQ related errors
 
     private final Stats statsClient = Stats.client();
 
     @Override
-    public InsertStatus push(Records records) {
+    public Status push(Records records) {
```

Comment:
could we'e feature toggle, so we can remove it, after it's rolled out for few prod services completely.

Also this breaks contract and tests are failing

Reply:
The `BqSink` is a `Sink` and the push method declared in `Sink` should return Status. Therefore BqSink should return `Status`.